shop/settings.py

```diff
@@ -1,53 +1,11 @@
 # -*- coding: utf-8 -*-
-from __future__ import unicode_literals
-
-from decimal import Decimal
-from django.conf import settings
-
-APP_LABEL = settings.SHOP_APP_LABEL  # mandatory setting without default
-
-DEFAULT_CURRENCY = getattr(settings, 'SHOP_DEFAULT_CURRENCY', 'EUR')
-
-MONEY_FORMAT = getattr(settings, 'SHOP_MONEY_FORMAT', '{symbol} {amount}')
-"""
-When rendering an amount of type Money, use this format. Possible placeholders are:
-{symbol}: This can be €, $, £, etc.
-{currency}: This can be EUR, USD, GBP, etc.
-{amount}: The localized amount .
-"""
-
-USE_TZ = True
 
-DECIMAL_PLACES = getattr(settings, 'SHOP_DECIMAL_PLACES', 2)
-"""
-Number of decimal places for the internal representation of a price. Not visible to the customer.
-"""
-
-CUSTOMER_SERIALIZER = getattr(settings, 'SHOP_CUSTOMER_SERIALIZER', 'shop.rest.defaults.CustomerSerializer')
-
-ORDER_ITEM_SERIALIZER = getattr(settings, 'SHOP_ORDER_ITEM_SERIALIZER', 'shop.rest.defaults.OrderItemSerializer')
-
-CART_MODIFIERS = getattr(settings, 'SHOP_CART_MODIFIERS', ('shop.modifiers.defaults.DefaultCartModifier',))
-
-VALUE_ADDED_TAX = getattr(settings, 'SHOP_VALUE_ADDED_TAX', Decimal('20'))
-
-ORDER_WORKFLOWS = getattr(settings, 'SHOP_ORDER_WORKFLOWS', ())
+from __future__ import unicode_literals
 
-ADD2CART_NG_MODEL_OPTIONS = getattr(settings, 'SHOP_ADD2CART_NG_MODEL_OPTIONS', "{updateOn: 'default blur', debounce: {'default': 500, 'blur': 0}}")
-EDITCART_NG_MODEL_OPTIONS = getattr(settings, 'SHOP_EDITCART_NG_MODEL_OPTIONS', "{updateOn: 'default blur', debounce: {'default': 500, 'blur': 0}}")
-"""
-Use ``SHOP_EDITCART_NG_MODEL_OPTIONS`` and ``SHOP_ADD2CART_NG_MODEL_OPTIONS`` to configure the
-update behavior when changing the quantity of a cart item. For more information refer to the
-AngularJS docs: https://code.angularjs.org/1.3.7/docs/api/ng/directive/ngModelOptions
-"""
 
-GUEST_IS_ACTIVE_USER = getattr(settings, 'SHOP_GUEST_IS_ACTIVE_USER', False)
-"""
-If ``SHOP_GUEST_IS_ACTIVE_USER`` is True, Customers which declared themselves as guests, may request
-a password reset, so that they can log into their account at a later time. The default is False.
-"""
+from shop import app_settings
 
-CACHE_DURATIONS = {
-    'product_html_snippet': 86400,
-}
-CACHE_DURATIONS.update(getattr(settings, 'SHOP_CACHE_DURATIONS', {}))
+import sys  # noqa
+settings = app_settings.__class__()
+settings.__name__ = __name__
+sys.modules[__name__] = settings
```

Comment:
Yes, I shamelessly copied that from allauth, they also point to a documentation there, maybe I should just add this comment as well:
```
# Ugly? Guido recommends this himself ...
# http://mail.python.org/pipermail/python-ideas/2012-May/014969.html
```

What is does, is basically to change the export value of a module.
So that a `from shop import app_settings` resolves to an instance of `AppSettings`.

Important sentence in the link:
```
This works because the import machinery is actively enabling this
hack, and as its final step pulls the actual module out of
sys.modules, after loading it. (This is no accident. The hack was
proposed long ago and we decided we liked enough to support it in the
import machinery.)
```